### ISSUE TITLE
Add DonorBox form iFrame into announcements

### ISSUE THREAD

**Issue by TheaSkaff** [inaturalist/inaturalist#4161]
(Please let me know if this doesn't belong here)

I'm moving this topic from a Slack convo (on Jun 11): 
- It seems like it's possible to embed the DonorBox form (the Support iNaturalist "campaign" form) into an announcement since it's an iFrame. 
- But it's a big (tall) form, so we're curious to see 1) what it looks like, and 2) potentially adjust the form to make it "shorter" e.g. remove "dedicate my donation" and/or "write us a comment".
- I imagine this might fit better in a dashboard announcement as opposed to welcome/index, but it wouldn't hurt to see what it looks like in both, before figuring out next steps, if that's possible. 
Thank you! - Thea

**Comment by loarie** [comment]
screenshot
![Image](https://github.com/user-attachments/assets/a3148151-4d4a-4b3d-983e-7fa3ab84d7cd)

**Comment by TheaSkaff** [comment]
Oh man, so tall! 

Is it helpful if I tinker with some of the CSS in a (different) cloned form? Trying to figure what's tweak-able & what's not ... the large font size in the amount buttons is weird.

We could probably get away with losing the header at least:
.tabs-header {
display:none;
}

Thank you for working on this!

**Comment by pleary** [comment]
We can adjust most of the CSS - font sizes, spacing, padding, etc. We probably can't remove disclaimers and their marketing. The tallest part of the form is a later screen asking for credit card info. Again we can change spacing a things, but we'll need to consider a lot of variables - users screen width, leaving room for validation errors to display. And even with changing styles, that CC form will still be quite tall. The more we change, the more cases we'll need to consider and we'll kind of be fighting their default styles. Note that when you adjust styles, I've found they will get cached so you may not see changes immediately on an embedded form outside the Donorbox site.

I did just discover a couple other options we might consider - one is a popup donation form, and one is a button they can add to the sidebar. See basic examples of both at https://donorbox.org/donation-forms. You can the pop-up version in donorbox by going to the campaign "Launchpad", clicking at the top "Pop-up Donation Form", then scrolling down to the preview "donate" button. To view the side panel version, on that same page enable "Preview Sticky Donate Button". The side panel style is also something we can do ourselves - its just a variant form of a button that opens the "Pop-up" form.

In my opinion that popup form would be far better for us than embedding the form in an iframe as we wouldn't be fighting their CSS and the form wouldn't take up real-estate on the page. Not sure how much the different options would appeal to users. Something to consider. If we wanted to continue with the iframe-embedded form and we wanted to change the styles, then I'd recommend it go through a design phase. If we wanted to go with the popup version, we could use that immediately without needing to mess too much with styles.

Let me know if there are other things you'd like to see done now with this ticket, or if we should close this and open other tickets in the future with additional requirements. If you want to view the form on staging, let me know and I can make a test announcement the same as what you see in these screenshots.

<img width="932" alt="Screenshot 2024-08-05 at 7 47 50 PM" src="https://github.com/user-attachments/assets/76762e77-4531-455f-96fc-8b41383fd474">

**Comment by TheaSkaff** [comment]
Super, thanks for the context Patrick, this all makes sense.  +1 to "The more we change, the more cases we'll need to consider and we'll kind of be fighting their default styles." 

We can definitely try the pop-up first as well as the sticky button, see how those perform--plus those have the advantage of allowing users to stay on the site, nice. At Wikimedia, we tested a pop-up vs iFrame and iFrame was the higher performer, but that said, this isn't a rush.

Also, while it's likely a dead end, I'll reach out to the folks at DonorBox to see if there's an easier way of embedding the iFrame without messing with a bunch of their CSS. It's sort of a derpy question, but like, "Do you have a smaller or "shorter" version of the form?

Next steps: How about we close this issue, I'll open a fresh issue for the pop-up and sticky button, I'll test those where I can, and I'll wait to see what DB says or if their engineers have any advice to share. And if that's not helpful, I'll open up a fresh issue and we can chip away at it. 

How does all that sound?

**Comment by pleary** [comment]
Thea reached out to DonorBox and ask about additional options for smaller donation forms to embed, and their response was they don't have any additional options and pointed us to the custom CSS feature we've been experimenting with here. I'll close this ticket as we found it is indeed possible to embed the iFrame into announcement. We tested a few alternatives including the pop-up form that can be opened in the same page using a button in an announcement.

As Thea said, if we want to continue to peruse embedding the donation form into an announcement, the next step would be to come up with specific style guidelines for how that should work. Once we have style guidelines, we can open a new issue in this project to implement the custom CSS needed to match those guidelines, and generate the iframe tag we'd need to embed in announcements to display that customized form
